torch/csrc/jit/python/pybind_utils.h

```diff
@@ -948,10 +950,71 @@ inline py::object runAndInsertCall(
   return toPyObject(std::move(stack.back()));
 }
 
+inline c10::optional<py::object> maybeTorchFunctionDispatch(
+    const py::object& callee,
+    const tuple_slice& args_no_self,
+    const py::kwargs& kwargs,
+    const c10::QualifiedName qualname) {
+  std::vector<py::handle> args_vec = {callee};
+  for (const auto& arg : args_no_self) {
+    args_vec.push_back(arg);
+  }
+  py::tuple args = py::cast(args_vec);
+
+  // Handle __torch_function__ dispatch
+  std::vector<py::handle> overloaded_args;
+  size_t total_arg_num = args.size() + kwargs.size();
+  for (size_t i = 0; i < args.size(); ++i) {
+    is_tensor_and_append_overloaded(args[i].ptr(), &overloaded_args);
+    is_tensor_list_and_append_overloaded(
+        args[i].ptr(),
+        &overloaded_args,
+        static_cast<int>(total_arg_num),
+        false /* throw_error */);
+  }
+  // NB: for kwargs, we cannot guarantee the order of appending
+  // is the same as the argument order in operator's schema.
+  // This is suboptimal, but should be fine. Later when we have
+  // better schema matching and argument parsing, we could
+  // match the operator in `operations` first, then the order will
+  // be guaranteed.
+  for (auto item : kwargs) {
+    is_tensor_and_append_overloaded(item.second.ptr(), &overloaded_args);
+    is_tensor_list_and_append_overloaded(
+        item.second.ptr(),
+        &overloaded_args,
+        total_arg_num,
+        false /* throw_error */);
+  }
+  if (overloaded_args.size() > 0) {
+    std::vector<py::object> overloaded_types;
+    overloaded_types.reserve(overloaded_args.size());
+    for (auto& oarg : overloaded_args) {
+      overloaded_types.push_back(
+          py::reinterpret_borrow<py::object>((PyObject*)Py_TYPE(oarg.ptr())));
+    }
```

Comment:
Types need to be weakly ordered -- Subclasses before superclasses. That invariant isn't satisfied here, unfortunately.

Reply:
Scratch that -- Looks like you aren't using `overloaded_types` at all.

Reply:
`overloaded_types` can be removed, I believe.